### PR TITLE
chore: fix linter warnings

### PR DESCRIPTION
**Pull request by jtieri**
This PR updates the CI jobs for linting and running tests to use current deps. It also fixes all the linter warnings from `golangci-lint`.

### CODE REVIEW

File: blockdb/messages_view_test.go [hunk 1]
```diff
@@ -9,17 +9,18 @@ import (
...
-	"github.com/strangelove-ventures/interchaintest/v8/testreporter"
-	"github.com/strangelove-ventures/interchaintest/v8/testutil"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/zap/zaptest"
 
 	"cosmossdk.io/math"
 
 	"github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/strangelove-ventures/interchaintest/v8"
```
Comment (Reecepbcups) [inline blockdb/messages_view_test.go:19]: add optional git hooks for vscode peeps

File: blockdb/sql_test.go [hunk 2]
```diff
@@ -112,7 +108,7 @@ func TestDB_Concurrency(t *testing.T) {
 			for {
 				if ctx.Err() != nil {
 					// Context was canceled; querying is finished.
-					return nil
+					return nil //nolint: nilerr
```
Comment (Reecepbcups) [inline blockdb/sql_test.go:111]: remove since ignoring this linter type now

File: chain/cosmos/chain_node.go [hunk 3]
```diff
@@ -191,7 +192,7 @@ func (tn *ChainNode) NewSidecarProcess(
 		VolumeName: v.Name,
 		ImageRef:   image.Ref(),
 		TestName:   tn.TestName,
-		UidGid:     image.UidGid,
+		UIDGID:     image.UIDGID,
```
Comment (Reecepbcups) [inline chain/cosmos/chain_node.go:195]: user breaking - lets until v10 - create ticket for this. Don't want people to have to refactor their test

Comment (jtieri) [reply chain/cosmos/chain_node.go]: reverted in 0e64f2cfb0a1a6863740fdb0d72c2f649321d6cd

**Review by Reecepbcups — APPROVED** [review]

**Pull request merged**